src/circuit/circuit_frame.rs

```diff
@@ -4644,15 +4643,117 @@ pub fn comparison_helper<F: LurkField, CS: ConstraintSystem<F>>(
 pub fn enforce_at_most_n_bits<F: LurkField, CS: ConstraintSystem<F>>(
     mut cs: CS,
     g: &GlobalAllocations<F>,
-    num: AllocatedNum<F>,
+    num_bits: Vec<Boolean>,
     n: usize,
 ) -> Result<(), SynthesisError> {
-    let num_bits = num.to_bits_le(&mut cs.namespace(|| "u64 remainder bit decomp"))?;
     let v = num_bits[n..255].to_vec();
     popcount(&mut cs.namespace(|| "add all MSBs"), &v, &g.false_num);
     Ok(())
 }
 
+#[derive(Copy, Clone)]
+pub enum UnsignedType {
+    U32,
+    U64,
+}
+
+// Convert from num to u32.
+pub fn to_unsigned_int<F: LurkField, CS: ConstraintSystem<F>>(
+    mut cs: CS,
+    g: &GlobalAllocations<F>,
+    maybe_unsigned: &AllocatedNum<F>,
+    size: UnsignedType,
+) -> Result<AllocatedNum<F>, SynthesisError> {
+
+    let p_bn =  match size {
+        UnsignedType::U32 => BigUint::pow(&BigUint::from_u32(2).unwrap(), 32),
+        UnsignedType::U64 => BigUint::pow(&BigUint::from_u32(2).unwrap(), 64),
+    };
+
+    let v = match maybe_unsigned.get_value() {
+        Some(v) => v,
+        None => F::zero(), //dummy
+    };
+    let field_bn = BigUint::from_bytes_le(v.to_repr().as_ref());
+
+    let (q_bn, r_bn) = field_bn.div_rem(&p_bn);
+
+    let q_num = allocate_unconstrained_bignum(&mut cs.namespace(|| "q"), q_bn)?;
+    let r_num = allocate_unconstrained_bignum(&mut cs.namespace(|| "r"), r_bn)?;
+
+    // a = b.q + r
+    let product = match size {
+        UnsignedType::U32 =>  mul(
+            &mut cs.namespace(|| "product(q,pow(2,32))"),
+            &q_num,
+            &g.power2_32_num,
+        )?,
+        UnsignedType::U64 =>  mul(
+            &mut cs.namespace(|| "product(q,pow(2,64))"),
+            &q_num,
+            &g.power2_64_num,
+        )?,
+    };
+    let sum = add(&mut cs.namespace(|| "sum remainder"), &product, &r_num)?;
+    let u_decomp = alloc_equal(
+        &mut cs.namespace(|| "check unsigned decomposition"),
+        &sum,
+        maybe_unsigned,
+    )?;
+    enforce_true(
+        &mut cs.namespace(|| "enforce decomposition"),
+        &u_decomp,
+    )?;
+
+    // enforce remainder range
+    let r_bits = r_num.to_bits_le(&mut cs.namespace(|| "u64 remainder bit decomp"))?;
+    let output = match size {
+        UnsignedType::U32 => {
+            enforce_at_most_n_bits(
+                &mut cs.namespace(|| "remainder u32 range"),
+                g,
+                r_bits,
+                32,
+            )?;
+            let output_alloc = AllocatedNum::alloc(&mut cs.namespace(|| "to_32"), || {
+                Ok(F::from_u32(v.to_u32_unchecked()).unwrap())
```

Comment:
Should this panic? 
if the field modulus is smaller than MAX_U32, the contract of `Primefield::from_repr` seems to be returning `None`:

https://github.com/zkcrypto/ff/blob/e853770f4843e47cd5e07330140b5bd1cdfc9f11/src/lib.rs#L264-L269

Should we not return an error here instead?